### PR TITLE
[End- to End Test code sprint] Add SEVIRI METEOSAT-8 and METEOSAT-11 to end-to-end testing

### PR DESCRIPTION
**Pull request by azadeh-gh**
(#765)

End-to-end testing for Spinning Enhanced Visible and InfraRed Imager (**SEVIRI**) from METEOSAT-8 and to METEOSAT-11

**New files include:**
- `parm/ioda/bufr2ioda//bufr2ioda_sevcsr.json`  : JSON containing data format, sensor/bufr2ioda_sevcsr.json, and satellite information
-  `ush/ioda/bufr2ioda/bufr2ioda_sevcsr.py` : bufr2ioda python code for converting SEVIRI METEOSAT-8 and METEOSAT-11 from BUFR to IODA 
- `parm/atm/obs/config/seviri_m08.yaml.j2`
- `parm/atm/obs/config/seviri_m11.yaml.j2`
- `parm/atm/obs/lists/gdas_prototype_3d.yaml.j2`

### CODE REVIEW

**Comment by RussTreadon-NOAA** [comment]
**Orion test**
Install 
- `parm/ioda/bufr2ioda/bufr2ioda_sevcsr.json`
- `ush/ioda/bufr2ioda/bufr2ioda_sevcsr.py`

in a working copy of `feature/gdas-validation`.   

Copy `seviri_m08.yaml` and `seviri_m11.yaml` from `parm/atm/obs/testing` to `/parm/atm/obs/config`.  Update both for use in g-w.  It's probable that the updated seviri yamls require additional changes for thinning, qc, etc.   This is just a starting point to allow `fv3jedi_var.x` execution.

Ensure seviri satbias, satbias_cov, and tlapse files are in appropriate `COMROT` directory.   Add `seviri_m08` and `seviri_m11` to `parm/atm/obs/lists/gdas_prototype_3d.yaml`

Execute prepatmiodaobs, atmanlinit, and atmanlrun.  The first two jobs run to completion.  SEVIRI dump files are written to the appropriate COM_OBS directory
```
-rw-r--r-- 1 rtreadon da   6353993 Nov 22 11:07 gdas.t00z.seviri_m11.tm00.nc
-rw-r--r-- 1 rtreadon da   4862394 Nov 22 11:07 gdas.t00z.seviri_m08.tm00.nc
```

atmanlrun begins but `fv3jedi_var.x` fails with the exception
```
313: Exception:         Reason: An exception occurred inside ioda while opening a variable.
313:    name:   MetaData/sensorAzimuthAngle
313:    source_column:  0
313:    source_filename:        /work2/noaa/da/rtreadon/gdas-validation/global-workflow/sorc/gdas.cd/ioda/src/engines/ioda/src/ioda/Has_Variables.cpp
313:    source_function:        ioda::Variable ioda::detail::Has_Variables_Base::open(const std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char>> &) const
313:    source_line:    108
313:
313: Exception: oops::Variational<FV3JEDI, UFO and IODA observations> terminating...
```
Notice that atms ioda format dump files contain MetaData `float sensorAzimuthAngle(Location)`.   This MetaData field is missing from seviri ioda format dump files.  _Should it be present?_

**Comment by BrettHoover-NOAA** [comment]
I changed `sensorZenithAngle` to `satelliteZenithAngle` in my AMV code to match changes that @emilyhcliu had made - its possible that there could be a variable name mismatch between different parts of the code-base as well

**Comment by RussTreadon-NOAA** [comment]
Refresh working copy of gdas-validation with SEVIRI updates.   Run prepatmiodaobs, atmanlinit, and atmanlrun.   `fv3jedi_var.x` terminated with

```
  0: OOPS_TRACE[0] ufo::Variable(conf) done
  0: OOPS_TRACE[0] ufo::Variable(config) start LocalConfiguration[root={name => brightnessTemperature , channels => (nil)}]
326: Exception: The key '.' is set to neither a string nor a map with the correct keys.
326: Bad Conversion: Cannot convert (nil) (Nil) to std::string
326: Bad Conversion: Cannot convert {name => brightnessTemperature , channels => (nil)} (OrderedMap) to std::string  (/work2/noaa/da/rtreadon/gdas-validation/global-workflow/sorc/gdas.cd/ufo/src/ufo/utils/parameters/ParameterTraitsVariable.cc +70 get)
378: Exception: The key '.' is set to neither a string nor a map with the correct keys.
378: Bad Conversion: Cannot convert (nil) (Nil) to std::string
378: Bad Conversion: Cannot convert {name => brightnessTemperature , channels => (nil)} (OrderedMap) to std::string  (/work2/noaa/da/rtreadon/gdas-validation/global-workflow/sorc/gdas.cd/ufo/src/ufo/utils/parameters/ParameterTraitsVariable.cc +70 get)
380: Exception: The key '.' is set to neither a string nor a map with the correct keys.
380: Bad Conversion: Cannot convert (nil) (Nil) to std::string
380: Bad Conversion: Cannot convert {name => brightnessTemperature , channels => (nil)} (OrderedMap) to std::string  (/work2/noaa/da/rtreadon/gdas-validation/global-workflow/sorc/gdas.cd/ufo/src/ufo/utils/parameters/ParameterTraitsVariable.cc +70 get)
```
Issue with input yaml file or code?   

Run directory is `/work2/noaa/stmp/rtreadon/RUNDIRS/gdas_eval_satwind_JEDI/gdasatmanl_00_seviri`.   Log file is `/work2/noaa/da/rtreadon/gdas-validation/comrot/gdas_eval_satwind_JEDI/logs/2021080100/seviri_gdasatmanlrun.log`

File: parm/atm/obs/config/seviri_m11.yaml [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+  where:
+  - variable:
+      name: GeoVaLs/land_area_fraction
+    minvalue: 0.99
+
+# Reject all channels over snow-dominant area
+- filter: RejectList
+  filter variables:
+  - name: brightnessTemperature
+    channels:
```
Comment (CoryMartin-NOAA) [inline parm/atm/obs/config/seviri_m11.yaml]: This I think is the issue @RussTreadon-NOAA is finding in his test

Comment (RussTreadon-NOAA) [reply parm/atm/obs/config/seviri_m11.yaml]: Good catch, @CoryMartin-NOAA .  I added `4-11` and reran atmanlinit & atmanlrun.   atmanlrun died with
```
 39: Exception: UserError: obs bias.covariance section missing from the YAML file
 41: Exception: UserError: obs bias.covariance section missing from the YAML file
  3: Exception: UserError: obs bias.covariance section missing from the YAML file

```
I know how to fix this.

**Comment by RussTreadon-NOAA** [comment]
Next `fv3jedi_var.x` error is

```
  0: OOPS_TRACE[0] UnstructuredInterpolator::UnstructuredInterpolator start
346: Exception: Bad parameter: /: Mandatory parameter 'obserr_bound_max' not found  (/work2/noaa/da/rtreadon/gdas-validation/glo\
bal-workflow/sorc/gdas.cd/oops/src/oops/util/parameters/RequiredParameter.h +135 deserialize)
347: Exception: Bad parameter: /: Mandatory parameter 'obserr_bound_max' not found  (/work2/noaa/da/rtreadon/gdas-validation/glo\
bal-workflow/sorc/gdas.cd/oops/src/oops/util/parameters/RequiredParameter.h +135 deserialize)
348: Exception: Bad parameter: /: Mandatory parameter 'obserr_bound_max' not found  (/work2/noaa/da/rtreadon/gdas-validation/glo\
bal-workflow/sorc/gdas.cd/oops/src/oops/util/parameters/RequiredParameter.h +135 deserialize)
```

File: parm/atm/obs/config/seviri_m08.yaml [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+      obserr_bound_latitude:
+        name: ObsFunction/ObsErrorFactorLatRad
+        options:
+          latitude_parameters: [0.0, 0.0, 0.0, 0.0]
+      obserr_bound_transmittop:
+        name: ObsFunction/ObsErrorFactorTransmitTopRad
+        channels: 4-11
+        options:
+          channels: 4-11
+      obserr_boundax: [ 2.0, 4.0, 3.5, 2.0, 2.0, 2.0, 2.0, 3.0 ]
```
Comment (CoryMartin-NOAA) [inline parm/atm/obs/config/seviri_m08.yaml]: @RussTreadon-NOAA here's your next issue, a typo. @azadeh-gh did you miss any commits when testing this that are not included in this PR?

**Comment by azadeh-gh** [comment]
Thank you @RussTreadon-NOAA and @CoryMartin-NOAA for your tests/comments. Sorry I made a mistake, I wanted to push these commits to my [feature/Azadeh_gdas-validation](https://github.com/NOAA-EMC/GDASApp/tree/feature/Azadeh_gdas-validation) branch 
I updated the python code and added two Seviri YAMLS to config but my "gdasatmanlinit" run fails with "Permission denied" error.
 I get permission denied for all 20210801.000000 files in
 /work2/noaa/da/gholoubi/gdas-validation/comrot/gdas_eval_seviri2_JEDI/gdas.20210731/18/model_data/atmos/restart

Any suggestions?

**Comment by CoryMartin-NOAA** [comment]
@azadeh-gh my guess is it is because you don't have `stmp` group access on Orion. Didn't we discuss this a couple of weeks ago (or was that someone else who also doesn't have it?)

**Comment by RussTreadon-NOAA** [comment]
Next `fv3jedi_var.x` error after fixing `obserr_bound_max`

```
194: Exception:         Reason: An exception occurred inside ioda while opening a variable.
194:    name:   MetaData/solarAzimuthAngle
194:    source_column:  0
194:    source_filename:        /work2/noaa/da/rtreadon/gdas-validation/global-workflow/sorc/gdas.cd/ioda/src/engines/ioda/src/ioda/Has_Variables.cpp
194:    source_function:        ioda::Variable ioda::detail::Has_Variables_Base::open(const std::__cxx11::basic_string<char, std\
::char_traits<char>, std::allocator<char>> &) const
194:    source_line:    108
194:
194: Exception: oops::Variational<FV3JEDI, UFO and IODA observations> terminating...
```

**Comment by azadeh-gh** [comment]
@CoryMartin-NOAA Yes, I still do not have stmp access (sent request to help desk), but I changed the DATAROOT from stmp to da, and can see other files in /work2/noaa/da/gholoubi/gdas-validation/comrot/gdas_eval_seviri2_JEDI/gdas.20210731/18/model_data/atmos/restart

**Comment by CoryMartin-NOAA** [comment]
@azadeh-gh can you point me to your logfile so I can take a look at where the permission denied problem is occurring?

**Comment by azadeh-gh** [comment]
@RussTreadon-NOAA solarAzimuthAngle and sensorAzimuthAngle are missing parameters in Seviri bufr file (bmiss in gsi read_seviri.f90).

based on obsmod.F90 :
def bmiss            - parameter to define missing value from bufr
! Set bufr missing value
#if def ibm_sp
  real(r_kind), parameter:: bmiss = 1.0e11_r_kind
#else
  real(r_kind), parameter:: bmiss = 1.0e9_r_kind
#endif

I used "fillval=float32_fill_value" for sensorAzimuthAngle is this correct? or I need to insert 1.0e11?

**Comment by azadeh-gh** [comment]
@CoryMartin-NOAA  here is the log file:
/work2/noaa/da/gholoubi/gdas-validation/comrot/gdas_eval_seviri2_JEDI/logs/2021080100/gdasatmanlinit.log

**Comment by CoryMartin-NOAA** [comment]
@azadeh-gh are you able to run the following commands?:
`cat /work2/noaa/da/gholoubi/gdas-validation/comrot/gdas_eval_seviri2_JEDI/gdas.20210731/18//model_data/atmos/restart/20210801.000000.coupler.res`
`cat /work2/noaa/da/cmartin/UFO_eval/data/para/output_ufo_eval_aug2021/2021080100/gdas.20210731/18/atmos/RESTART/20210801.000000.coupler.res` ?

**Comment by RussTreadon-NOAA** [comment]
> @RussTreadon-NOAA solarAzimuthAngle and sensorAzimuthAngle are missing parameters in Seviri bufr file (bmiss in gsi read_seviri.f90).
> 
> based on obsmod.F90 : def bmiss - parameter to define missing value from bufr ! Set bufr missing value #if def ibm_sp real(r_kind), parameter:: bmiss = 1.0e11_r_kind #else real(r_kind), parameter:: bmiss = 1.0e9_r_kind #endif
> 
> I used "fillval=float32_fill_value" for sensorAzimuthAngle is this correct? or I need to insert 1.0e11?

Great question which prompts two additional questions
1. What do we encode (data type and range) for other satellite_sensors which have a valid solarAzimuthAngle?   
2. Does UFO/IODA have something akin to the GSI `bmiss`?  If so, what do UFO/IODA expect when a metadata field has a missing value?

**Comment by azadeh-gh** [comment]
@CoryMartin-NOAA No, I get permission denied

**Comment by CoryMartin-NOAA** [comment]
@azadeh-gh fascinating. I wonder why, you are in the `da` group...

**Comment by azadeh-gh** [comment]
@CoryMartin-NOAA I can run :  cat /work2/noaa/da/gholoubi/gdas-validation/comrot/gdas_eval_seviri2_JEDI/gdas.20210731/18//model_data/atmos/restart/20210801.020000.coupler.res

I only get permission denied for 20210801.000000 files

**Comment by CoryMartin-NOAA** [comment]
@azadeh-gh thanks I'll check and see if there are any differences

**Comment by CoryMartin-NOAA** [comment]
@azadeh-gh ha!
```
-rw-r----- 1 cmartin stmp 300 Apr 25  2022 /work2/noaa/da/cmartin/UFO_eval/data/para/output_ufo_eval_aug2021/2021080100/gdas.20210731/18/atmos/RESTART/20210801.000000.coupler.res
-rw-r----- 1 cmartin da   300 Apr 25  2022 /work2/noaa/da/cmartin/UFO_eval/data/para/output_ufo_eval_aug2021/2021080100/gdas.20210731/18/atmos/RESTART/20210801.010000.coupler.res
-rw-r----- 1 cmartin da   300 Apr 25  2022 /work2/noaa/da/cmartin/UFO_eval/data/para/output_ufo_eval_aug2021/2021080100/gdas.20210731/18/atmos/RESTART/20210801.020000.coupler.res
-rw-r----- 1 cmartin da   300 Apr 25  2022 /work2/noaa/da/cmartin/UFO_eval/data/para/output_ufo_eval_aug2021/2021080100/gdas.20210731/18/atmos/RESTART/20210801.030000.coupler.res
```
I'll chgrp to da

**Comment by azadeh-gh** [comment]
@CoryMartin-NOAA Thank you!

**Comment by RussTreadon-NOAA** [comment]
@azadeh-gh , you [mentioned]([float32_fill_value]) using 1.e11 for `solarAzimuthAngle`.  I modified my working copy of `bufr2ioda_sevcsr.py` to do this.    `fv3jedi_var.x` ran past the `MetaData/solarAzimuthAngle` error, but it still aborted.  The new traceback is
```
  9: Exception:         source_column:  0
  9:    source_filename:        /work2/noaa/da/rtreadon/gdas-validation/global-workflow/sorc/gdas.cd/ioda/src/engines/ioda/include/ioda/Variables/Variable.h
  9:    source_function:        Variable_Implementation ioda::detail::Variable_Base<Variable_Implementation>::read(gsl::span<DataType>, const ioda::Select\
ion &, const ioda::Selection &) const [with DataType = int; Marshaller = ioda::detail::Object_Accessor_Regular<int, int>; TypeWrapper = ioda::Types::GetType_Wrapper<int, 0>; Variable_Implementation = ioda::Variable]
  9:    source_line:    539
  9:
  9: Exception: oops::Variational<FV3JEDI, UFO and IODA observations> terminating...
```
I'm not confident that I added `solarAzimuthAngle` correctly.

While running gdasprepatmiodaobs I noticed the follow warning messages from `bufr2ioda_sevcsr.py`
```
Warning: Query String */SDTB/TMBRST didn't apply to subset NC021043
Warning: Query String */TMBRST didn't apply to subset NC021043
Warning: Query String */SCCF didn't apply to subset NC021043
Warning: Query String */NCLDMNT didn't apply to subset NC021043
Warning: Query String */SIID[1] didn't apply to subset NC021043
```
Are these warning messages expected?

**Comment by RussTreadon-NOAA** [comment]
@azadeh-gh , is there any way in which I can help with troubleshooting?

**Comment by azadeh-gh** [comment]
@RussTreadon-NOAA Thanks Russ, I added  solaziang=np.array([float32_fill_value], dtype=np.float32) for solarAzimuthAngle.
I'm running the JEDI 3D-Var, and found other parameters that need to be added.

**Comment by RussTreadon-NOAA** [comment]
Thank you @azadeh-gh for systematically identifying and resolving issues with seviri processing.

**Comment by emilyhcliu** [comment]
@azadeh-gh needs to fix coding norm errors

**Comment by emilyhcliu** [comment]
@azadeh-gh Thanks for bringing the branch updated with the develop!   I am about to run the test (ORION will be off tomorrow).

**Comment by nicholasesposito** [comment]
I think this needs a new merge. It says there are 235 files changed.

**Comment by emilyhcliu** [comment]
@azadeh-gh There are 235 files changed.  This is odd.  

> I think this needs a new merge. It says there are 235 files changed.

I just found the same thing. @azadeh-gh you changed 5 files, I remembered.

**Comment by emilyhcliu** [comment]
> @azadeh-gh There are 235 files changed. This is odd.
> 
> > I think this needs a new merge. It says there are 235 files changed.
> 
> I just found the same thing. @azadeh-gh you changed 5 files, I remembered.

I changed the base from feature/gdas-validation to develop. 
Now, the changes are down from 235 to 5.

**Comment by RussTreadon-NOAA** [comment]
Please update `feature/Azadeh_gdas-validation` with recent changes from GDASApp `develop`.  Note that file `parm/atm/obs/lists/gdas_prototype_3d.yaml` conflicts with `develop`.

**Comment by RussTreadon-NOAA** [comment]
@azadeh-gh , [`feature/Azadeh_gdas-validation`](https://github.com/NOAA-EMC/GDASApp/tree/feature/Azadeh_gdas-validation) is 47 commits behind `develop`.  Updating with `develop` is straightforward with the following caveat.

File `parm/atm/obs/lists/gdas_prototype_3d.yaml` no longer exists in `develop`.  The replacement file is named `parm/atm/obs/lists/gdas_prototype_3d.yaml.j2`.  Your older copy of the file activates two observation yamls
```
- !INC ${OBS_YAML_DIR}/seviri_m08.yaml
- !INC ${OBS_YAML_DIR}/seviri_m11.yaml
```
These yamls are not listed in `gdas_prototype_3d.yaml.j2`.  Adding them is not as simple as adding the above serviri yaml files.  The above seviri yaml file need to be converted to jinja2 templates and then `gdas_prototype_3d.yaml.j2` updated to include the seviri jinja2 templates.

Do you want assistance with this?

**Comment by azadeh-gh** [comment]
@RussTreadon-NOAA Do I need to update the seviri YAMLs to convert to jinja2?

**Comment by RussTreadon-NOAA** [comment]
Yes, the seviri yamls need to be updated to jinja2 templates

- replace `$( foo )` with `{{ foo }}`
- add single quotes around filenames on `obsfile` lines
- add two blank spaces at the start of each line in the yaml file.  shifting comments is options.  The one exception to this is the first line.  `obs space` shifts over two spaces as `- obs space` 
- change file suffix from `.yaml` to `.yaml.j2`

Beware:  If you change the file suffix via `git mv` and in the same commit modify the _new_ jinja2 template, the previous history of the yaml is lost.   git treats the combination of these two actions as creation of a new file  I think two commits will preserve the history.  One commit modifies the file.  Another commit moves the file from _filename1_ to _filename2_.

**Comment by azadeh-gh** [comment]
@RussTreadon-NOAA Thank you. I updated the YAMLS to jinja2.

**Comment by emilyhcliu** [comment]
Reviewers.  Please look at the [T2O Issue #99](https://github.com/NOAA-EMC/JEDI-T2O/issues/99) related to this PR for test results.

@azadeh-gh  I have two comments:
1. There are many plots in the [issue #99](https://github.com/NOAA-EMC/JEDI-T2O/issues/99)
I think it would be good to have a short summary of your validation results here.  
2.  Please add error handling in the Python converter for missing input files or missing subsets
     - the need to add the error handling was identified by @RussTreadon-NOAA  (see [PR #937](https://github.com/NOAA-EMC/GDASApp/pull/937) for more infomation)
     - here is the example of error handling added in the converter
    
Screenshot 2024-03-21 at 11 34 43 PM

**Comment by RussTreadon-NOAA** [comment]
@azadeh-gh , the conflict in `parm/atm/obs/lists/gdas_prototype_3d.yaml.j2` is easily resolved.  `develop` adds Emily's ozone work to the obs list.  Your branch has seviri.  Both can ozone and seviri can be in the obs list.

**Comment by azadeh-gh** [comment]
SEVIRI-QC-Flowchart pptx

### Summary of validation results

**No Thinning without QC (UFO vs. GSI) - Channel 5 Time Level 3** 

ufo_seviri_m08_omf_bc_channel5_time1
gsi_seviri_m08_omf_gsi_bc_channel5_time1

**No Thinning with QC: UFO vs. GSI - Channel 5  Time Level 1**

ufo_seviri_m08_omf_qc_channel5_time1
gsi_seviri_m08_omf_gsi_qc_channel5_time1

GSI_OmF_Seviri08_ch5_Thinned_Passed_QCs
JEDI_OmF_Seviri08_CH5_Thinned_PassedQCs

**Here are the channel statistics plots (#obs, bias/std for OMF and final obs error) for JEDI and GDAS runs (Thinned)**

**O-F Bias/Standard Deviation**
JEDI            |  GDAS (original: ptime=0 & ithin_time=1)
:-------------------------:|:-------------------------:
 | 

**Final Obs Error**
JEDI            |  GDAS (original: ptime=0 & ithin_time=1)
:-------------------------:|:-------------------------:
|

**Number Obs Passed QC**
JEDI            |  GDAS (original: ptime=0 & ithin_time=1)
:-------------------------:|:-------------------------:
|

**O-F Bias/Standard Deviation**
JEDI            |  GDAS (ptime=0 & ithin_time=2)
:-------------------------:|:-------------------------:
 | 

**Final Obs Error**
JEDI            |  GDAS (ptime=0 & ithin_time=2)
:-------------------------:|:-------------------------:
 | 

**Number Obs Passed QC**
JEDI            |  GDAS (ptime=0 & ithin_time=2)
:-------------------------:|:-------------------------:
|

**Comment by azadeh-gh** [comment]
> @azadeh-gh , the conflict in `parm/atm/obs/lists/gdas_prototype_3d.yaml.j2` is easily resolved. `develop` adds Emily's ozone work to the obs list. Your branch has seviri. Both can ozone and seviri can be in the obs list.

Thank you Russ. Updated the `parm/atm/obs/lists/gdas_prototype_3d.yaml.j2` to have both SEVIRI and Ozone in the obs list.

**Comment by azadeh-gh** [comment]
> Reviewers. Please look at the [T2O Issue #99](https://github.com/NOAA-EMC/JEDI-T2O/issues/99) related to this PR for test results.
> 
> @azadeh-gh I have two comments:
> 
> 1. There are many plots in the [issue #99](https://github.com/NOAA-EMC/JEDI-T2O/issues/99)
>    I think it would be good to have a short summary of your validation results here.
> 2. Please add error handling in the Python converter for missing input files or missing subsets
>    
>    * the need to add the error handling was identified by @RussTreadon-NOAA  (see [PR #937](https://github.com/NOAA-EMC/GDASApp/pull/937) for more infomation)
>    * here is the example of error handling added in the converter
> 
> Screenshot 2024-03-21 at 11 34 43 PM

Thank you for your suggestions. 
1. Added summary of our evaluation results from the [Issue #99](https://github.com/NOAA-EMC/JEDI-T2O/issues/99) here.
2. Updated my python converter and added error handling for missing input files or missing subsets.

**Comment by RussTreadon-NOAA** [comment]
@azadeh-gh , I went to test `feature/Azadeh_gdas-validation` for a near real-time case (2024022400).  The `sevcsr` bufr dump file is zero length.  Your code noted this and properly exited. That's good.   A check of the gdas directory in the GDA shows all `gdas.t*z.sevcsr.tm00.bufr_d` files for 2024 are, thus far, zero length.

Do we know why all 2024 `sevcsr` dump files are zero length?  Will the data return?

**Review by RussTreadon-NOAA — COMMENTED** [review]
Changes look reasonable.  Unable to test in JEDI ATM CI case.  Bufr dump file `gdas.t00z.sevcsr.tm00.bufr_d` is zero length.

```
-rw-r--r-- 1 wkolczyn global 0 Feb 24 05:53 /work/noaa/rstprod/dump/gdas.20240224/00/atmos/gdas.t00z.sevcsr.tm00.bufr_d
```

**Comment by ADCollard** [comment]
@RussTreadon-NOAA We cannot do a real time test of SEVIRI as the CSR stream has been discontinued.   The same information is in the ASR stream, so the next step after this is to create an ASR converter.

**Review by ADCollard — APPROVED** [review]
The results in [JEDI-T2O Issue 99](https://github.com/NOAA-EMC/JEDI-T2O/issues/99) look good.  Approve.

**Comment by RussTreadon-NOAA** [comment]
Thank you @ADCollard for explaining what I am seeing.

**Pull request merged**